### PR TITLE
feat(python): add wait helpers

### PR DESCRIPTION
**Pull request by shortcuts**
## 🧭 What and Why

🎟 JIRA Ticket: https://algolia.atlassian.net/browse/DI-1737

### Changes included:

add wait method helpers to the Python client, tested in the playground for now it seems to work well

### CODE REVIEW

File: clients/algoliasearch-client-python/algoliasearch/http/helper.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    def __call__(self) -> int:
+        return 0
+
+
+class RetryTimeout(Timeout):
+    def __call__(self, retry_count: int) -> int:
+        return min(retry_count * 0.2, 5)
+
+
+async def create_iterable_promise(
```
Comment (millotp) [inline clients/algoliasearch-client-python/algoliasearch/http/helper.py:19]: is there a concept of `promise` in python ? maybe it should be called `create_iterable_task`

**Review by millotp — COMMENTED** [review]
really cool !

File: clients/algoliasearch-client-python/algoliasearch/http/helper.py (hunk 1 above)
Comment (shortcuts) [reply clients/algoliasearch-client-python/algoliasearch/http/helper.py:19]: actually I'll just name it create_iterable, the async response type is implied by the definition

**Review by millotp — APPROVED** [review]
perfect !

**Pull request merged**